Valeriia/Movies_database_project/src/components/SearchForm/SearchForm.tsx

```diff
@@ -21,12 +21,8 @@ const SearchForm = (props: ISearchFormProps) => {
     };
     const { filterBy, searchTerm } = query;
     setFilterBy(filterBy || "title");
-    setSearchTerm(searchTerm);
-  }, [props.location.search]);
-
-  const onChangeHadler = (searchTerm: string) => {
-    setSearchTerm(searchTerm);
-  };
+    setSearchTerm(searchTerm || "");
+  }, [props.history.action, props.location.search]);
```

Comment:
You've added `props.history.action ` as dependency, but you don't use it inside of useEffect. Probably you forgot to wrap your code in ` if (props.history.action !=== 'PUSH') ` condition